### PR TITLE
Don’t propagate bound values to occupied outputs

### PR DESCRIPTION
**Pull request by evgenykochetkov**
No issue 😬

See [this xodball](https://gist.github.com/evgenykochetkov/ce828868a657f7bfd529631a96f88cac) for a minimal example

### CODE REVIEW

File: packages/xod-project/src/flatten.js [hunk 1]
```diff
@@ -437,14 +437,26 @@ const getValueToBind = R.curry((nodesById, linksChain) => {
...
+      .chain((bottomBoundValue) => {
+        const pinKey = Link.getLinkOutputPinKey(R.last(linksChain));
+        const topNode = R.last(outputNodesFromLinkChain);
+        const maybeBoundValue = Node.getBoundValue(pinKey, topNode);
+
+        // if some value is already bound to the
+        // top node's output — don't overwrite it
+        return maybeBoundValue.isJust
+          ? Maybe.Nothing()
+          : Maybe.Just(bottomBoundValue);
```
Comment (nkrkv) [inline packages/xod-project/src/flatten.js]: Use `foldMaybe`? It would survive possible (?) `Maybe` lib replacement.

Comment (evgenykochetkov) [reply packages/xod-project/src/flatten.js]: Done 👍🏻

**Comment by evgenykochetkov** [comment]
🤔
It's not so simple.
We also need to consider wether the processed terminal nodes were input-terminals or output-terminals before conversion to `xod/internal/terminal-*`

**Comment by evgenykochetkov** [comment]
Updated the fix.
@nkrkv @brusherru check it out again please

**Pull request merged**